### DISCUSSION TITLE
Suggestion to simplify the use of this fast moving library

### DISCUSSION THREAD

**Discussion by philippejadin** [pschatzmann/arduino-audio-tools#1520] [Ideas]
This is purely based on my experience, so take this with a grain of salt.
I got quite a few compile "errors" due to the library changing quickly. For example  setNotifyAudioChange -> addNotifyAudioChange
Since (I try) to use platformio, it will autoupdate the dependencies when a full clean is done, or whenever I work on a new computer.

This make it harder to have a reproductibe environement. I now pin my dependencies so I guess it will be less of a problem, but imvvho, if you worked on a separate dev branch that you merge with master only when you release a new version it would help people like me, and might also have the good side effect to reduce support.

If it's a bad idea for wathever reason I didn't see, discard it all ;-)

**Comment by pschatzmann** [comment]
I see your point and I think specifying the revision when you define the project is a good choice, if you want to have a stable environment.

The master branch contains the latest corrections and changes that will make it to the next release. 

Before I create a new release, I would like to have the confirmation that I did not miss anything in my tests, so I am usually waiting a bit before creating a new release. 

I usually try to avoid breaking changes, but for your mentioned topic, I had the feeling that I was already waiting too long and you can set USE_OBSOLETE to true if you will want to use the obsolete API.
